onnxscript/rewriter/_ir_utils.py

```diff
@@ -78,23 +78,34 @@ def get_numpy_value(val: ir.Value | None) -> np.ndarray | None:
     return None
 
 
-def get_singleton_value(val: ir.Value | None, rank: int | None = None):
+def get_singleton_value(val: ir.Value | None, rank: int | Sequence[int] | None = None):
```

Comment:
changes to the helper lgtm. I think we can simplify the pattern though.

Reply:
Simplified the pattern for now. (More cleanup required elsewhere since we have test-cases that use 0D, and also the ORT version of these patterns.)